### PR TITLE
[EAGLE-3920] eval utils

### PR DESCRIPTION
**Pull request by phatvo9**
### What
Add some helper functions to handle evaluation data processing and visualize metric charts.

### How
- Generate summary tables for individual concepts and the overall total.
- visualize 
  - ROC
  - PR
  - confusion matrix

Example usage:
```python
models = [Model(url1), Model(url2)]
compare = EvalResultCompare(models, datasets=Dataset(...))
compare.all(output_dir)
```

### CODE REVIEW

**Review by sanjaychelliah — COMMENTED** [review]
Great addition @phatvo9 🔥 

Added some comments..

**Review by sanjaychelliah — COMMENTED** [review]
Also can we add a snippet of example code in main README.md

File: clarifai/client/model.py [hunk 1]
```diff
@@ -576,3 +576,180 @@ def __str__(self):
...
+            user_id=dataset_user_id or self.auth_helper.user_id,
+        ),
+        eval_info=eval_info_params,
+    )
+    request = service_pb2.PostEvaluationsRequest(
+        user_app_id=self.user_app_id,
+        eval_metrics=[eval_metric],
+    )
+    response = self._grpc_request(self.STUB.PostEvaluations, request)
+    if response.status.code != status_code_pb2.SUCCESS:
```
[the remark this answers is not in the document]
Comment (phatvo9) [reply clarifai/client/model.py:670]: The model will start evaluating when status is success, I added log..

File: clarifai/utils/evaluation.py [hunk 2]
```diff
@@ -0,0 +1,884 @@
...
+      for each in list_eval_outputs:
+        if each.status.code == status_code_pb2.MODEL_EVALUATED:
+          _dataset = each.ground_truth_dataset
+          # if version_id is empty -> get latest eval result of dataset,app,user id
+          if app_id == _dataset.app_id and dataset_id == _dataset.id and user_id == _dataset.user_id and (
+              not version_id or version_id == _dataset.version):
+            self.eval_data.append(each)
+            _is_found = True
+            break
+      assert _is_found, f"Model {self.model.model_info.name} in app {self.model.model_info.app_id} is not evaluated successfully with dataset {dataset_assert_msg}"
```
[the remark this answers is not in the document]
Comment (phatvo9) [reply clarifai/utils/evaluation.py]: Thank you. What a great idea, updated the code.

File: clarifai/utils/evaluation.py
[the remark this answers is not in the document]
Comment (phatvo9) [reply clarifai/utils/evaluation.py:1]: Thanks, I updated it

**Review by sanjaychelliah — APPROVED** [review]
LGTM 👍

**Review by ackizilkale — APPROVED** [review]
looks great. looking forward to using this!

**Pull request merged**